externs/shaka/abr_manager.js

```diff
@@ -103,6 +103,14 @@ shaka.extern.AbrManager = class {
    */
   playbackRateChanged(rate) {}
 
+  /**
+   * Set media element.
+   *
+   * @param {HTMLMediaElement} mediaElement
+   * @exportDoc
+   */
+  setMediaElement(mediaElement) {}
+
```

Comment:
The configuration already supports `restrictions.maxWidth` and `restrictions.maxHeight`, why does the ABR manager need a direct reference to the video element? Couldn't the restriction simply be applied at the player level?

Reply:
This is necessary for a future enhancement that will take dropped frames into account also.

Also note that the ResizeObserver has to take into account the element to be observed. If the video element is not passed, the ResizeObserver could not be done in the ABR and would have to be done externally.

I really think that this part is core to be able to develop this functionality.

@joeyparrish , what do you think?

Note: about the restrictions I will answer you in another thread

Reply:
If you're planning a feature to account for dropped frames, then I think having the element makes sense.  You could also imagine an AbrManager plugin that used buffer fullness instead of bandwidth, so this could have many uses.

I initially was going to recommend you accept the video element through the constructor, and let the factory supply it.  The default could be something like `() => new shaka.abr.SimpleAbrManager(player.getMediaElement())`.  But perhaps that's too obscure.  If a plugin doesn't care about the media element, it could simply have an empty setMediaElement method.

Reply:
For the moment then I will not make any changes to this part.